### PR TITLE
Support load state dict form `inference model` format save result

### PR DESCRIPTION
**Pull request by chenwhql**
### PR types

New features
### PR changes

APIs
### Describe

【兼容升级】This PR supprot loading `state_dict` from `save_inference_model` save result.

This is an important feature, and we often see related problems in issues. After we switched from static graph mode to dynamic graph mode, many users rewritten the dynamic graph model code and needed to load from the result of `save_inferenece_model` in original static graph model, such as: https://github.com/PaddlePaddle/Paddle/issues/26519

API changes:
- `paddle.load(model_path, keep_name_table=False)` -> `paddle.load(model_path, configs=None)`: add decorator `deprecate_keep_name_table` for compatibility
- `paddle.jit.SaveLoadConfig` -> `paddle.SaveLoadConfig`: because `paddle.load` also use it, move it to paddle module, `SaveLoadConfig` is a class for two save format's compatibility, not only a jit related class
- add `paddle.SaveLoadConfig.keep_name_table`: for compatible with old `paddle.load(model_path, keep_name_table=False)`

After this PR, the Paddle 2.0 Save/Load inferfaces are like follow:

image
image

This PR creates the `green line` in graph.

#### Doc change

related PR: https://github.com/PaddlePaddle/FluidDoc/pull/2531

image
image

### CODE REVIEW

**Review by phlrain — APPROVED** [review]

**Review by lanxianghit — APPROVED** [review]

**Review by saxon-zh — APPROVED** [review]
LGTM

File: python/paddle/fluid/dygraph/checkpoint.py [hunk 1]
```diff
@@ -100,41 +134,55 @@ def save_dygraph(state_dict, model_path):
 
 # TODO(qingqing01): remove dygraph_only to support loading static model.
 # maybe need to unify the loading interface after 2.0 API is ready.
-#@dygraph_only
-def load_dygraph(model_path, keep_name_table=False):
+# @dygraph_only
+@deprecate_keep_name_table
+def load_dygraph(model_path, configs=None):
```
Comment (XiaoguangHu01) [inline python/paddle/fluid/dygraph/checkpoint.py:139]: configs -> config
这里建议用单数，因为SaveLoadConfig是单数形式
config = paddle.SaveLoadConfig()
可以后续统一再改下

File: python/paddle/fluid/dygraph/jit.py [hunk 2]
```diff
@@ -600,6 +602,54 @@ def separate_params(self, value):
...
+            
+                paddle.disable_static()
+
+                linear = paddle.nn.Linear(5, 1)
+
+                state_dict = linear.state_dict()
+                paddle.save(state_dict, "paddle_dy")
+
+                configs = paddle.SaveLoadConfig()
+                configs.keep_name_table = True
```
Comment (XiaoguangHu01) [inline python/paddle/fluid/dygraph/jit.py:629]: keep_name_table默认设置为True的话，会有什么问题吗？
不是很了解背景

Comment (chenwhql) [reply python/paddle/fluid/dygraph/jit.py:629]: 也没有什么风险，只是用户载入的state_dict里面会多一些额外信息，这些信息对用户一般没有帮助，保留这个选项主要是为了兼容，怕有同学用利用了旧版实现里面的这个信息

File: python/paddle/fluid/dygraph/checkpoint.py (hunk 1 above)
Comment (chenwhql) [reply python/paddle/fluid/dygraph/checkpoint.py:139]: 好的，在下一个PR中修改

Comment (chenwhql) [reply python/paddle/fluid/dygraph/checkpoint.py:139]: 这里能否将SaveLoadConfig改为SaveLoadConfigs，这样兼容性更好处理一些，因为还涉及到jit.save和jit.load接口中的configs参数

File: python/paddle/fluid/dygraph/jit.py (hunk 2 above)
Comment (chenwhql) [reply python/paddle/fluid/dygraph/jit.py:629]: 或许我们可以先将这个参数作为内部参数，不向用户公开

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**